### ISSUE TITLE
[1.20.1] Hoglins crash on Heavenly Challenger

### ISSUE THREAD

**Issue by Dragonir44** [Aureljz/WhenDungeonsArise--Forge#26]
Hello, I create this problem following a crash of my server.

The crash always happens when we find a hoglin ridden by a wither skeleton on a Heavenly Challenger in the end.

Here's a link to the crash report: https://gist.github.com/Dragonir44/cc297951a85bf085a96dfbfc874a183d

In this crash report, you can find the hoglin causing the crash, but hoglin spawned naturally in the nether do not crash the server.
This crash does not appear in the single-player game.

Server info : 
- Forge version: 1.20.1-47.2.1
- Mod version: 1.20.1-2.1.57-release (tested in dev 1.20.1-2.1.56.1-beta and same things happen)

I'm here if you need further clarification.
Have a nice day!

**Comment by Aureljz** [comment]
WDA doesn't modify entity AI, it just uses what's available in vanilla minecraft. For this reason, it's unlikely this mod is causing a ticking entity crash.

`at net.minecraft.world.entity.ai.Brain.m_21957_(Brain.java:432) ~[server-1.20.1-20230612.114412-srg.jar%23449!/:?] {re:mixin,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B,pl:mixin:APP:moonlight-common.mixins.json:accessor.BrainAccessor,pl:mixin:A}`

I suspect it could be related to Moonlight mixins, as shown here: _moonlight-common.mixins.json:accessor.BrainAccessor_

**Comment by Dragonir44** [comment]
Ok, thanks for your answer, I'm not very good at reading crash logs (especially java errors). I'll keep an eye out for updates to moonlight (since it's a library, it may take a while to get it up to date).

Thanks again for your help :)

**Comment by Dragonir44** [comment]
I'm back after a few tests. I deactivated moonlight as well as Supplementaries which depends on it and it didn't change anything.
I also deactivated all the mods I saw in the crash report, but to no avail.
I tried to see if I had the same problem on the Heavenly Challengers in overworld and it doesn't seem to be the case.
The only mod I haven't been able to disable is The Bumblezone (disabling it prevents my server from launching because of missing tags that it can't ignore, but that's another problem).
I'll give it a try on a map without it, just to see.
